File: 1.two-sum.go
// @leet start
func twoSum(nums []int, target int) []int {
	remainingMap := make(map[int]int, 0)

	for i:= 0; i <= len(nums); i++ {
		remaining := target - nums[i]
		v, ok := remainingMap[remaining]
		if !ok {
			remainingMap[nums[i]] = i
			continue
		}
		return []int{v, i}
	}
	return []int{}
}
// @leet end

File: 1.two-sum.js
// @leet start
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
	let map = {}
	for (let i in nums) {
		num = nums[i]
		console.log(`i:${i} (${num})`)

		remaining = target - num
		if (map[remaining] == undefined) {
			map[num] = i
			continue
		}
		return [parseInt(map[remaining]), parseInt(i)]
	}
	return [0, 0]
};
// @leet end
